external/kinesis-asl/src/main/scala/org/apache/spark/streaming/kinesis/SerializableCredentialsProvider.scala

```diff
@@ -83,3 +84,146 @@ private[kinesis] final case class STSCredentialsProvider(
     }
   }
 }
+
+@InterfaceStability.Stable
+object SerializableCredentialsProvider {
+  /**
+   * Builder for [[SerializableCredentialsProvider]] instances.
+   *
+   * @since 2.2.0
+   */
+  @InterfaceStability.Stable
+  class Builder {
+    private var awsAccessKeyId: Option[String] = None
+    private var awsSecretKey: Option[String] = None
+    private var stsRoleArn: Option[String] = None
+    private var stsSessionName: Option[String] = None
+    private var stsExternalId: Option[String] = None
+
+
+    /**
+     * Sets the AWS access key ID when using a basic AWS keypair for long-lived authorization
+     * credentials. A value must also be provided for the AWS secret key.
+     *
+     * @param accessKeyId AWS access key ID
+     * @return Reference to this [[SerializableCredentialsProvider.Builder]]
+     */
+    def awsAccessKeyId(accessKeyId: String): Builder = {
```

Comment:
I'll rework this builder to take multiple arguments for the long-lived keypair and STS